spec/unit/file_system_spec.rb

```diff
@@ -134,6 +134,36 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
     end
   end
 
+  context "read should allow an encoding to be specified" do
+    # First line of Rune version of Rune poem at http://www.columbia.edu/~fdc/utf8/
+    # characters chosen since they will not parse on Windows with codepage 437 or 1252
+    let (:rune_utf8) { "\u16A0\u16C7\u16BB" } # 'ᚠᛇᚻ'
+
+    before :each do
+      # verify Ruby has treated string as UTF-8
+      expect(rune_utf8.encoding).to eq(Encoding::UTF_8)
```

Comment:
:+1: 

And to add to this, the `\u` is the escape for Unicode codepoints 0000 through FFFF, and was introduced in 1.9.  Strings using `\u` are always encoded in UTF8 per https://books.google.com/books?id=jcUbTcr5XWwC&pg=PA50&lpg=PA50&dq=ruby+%5Cu&source=bl&ots=fJFkra8zcB&sig=uyxrADkv11eAYEBKjjq6E52FOA8&hl=en&sa=X&ved=0ahUKEwjc5eDhus3KAhUQ6GMKHav8B5QQ6AEITTAH#v=onepage&q=ruby%20%5Cu&f=false

More info at
https://leejava.wordpress.com/2009/03/11/unicode-escape-in-ruby/